Tools/automation.ps1

```diff
@@ -0,0 +1,791 @@
+                Write-Host "    Creating base Azure Function object."
+                $Result = New-AzResourceGroupDeployment -ResourceGroupName $AzResourceGroup -TemplateFile $Object.TemplatePath -TemplateParameterFile $Object.ParameterPath -Mode Incremental -ErrorAction SilentlyContinue
+            }
+    
+            ## Creates the Azure Resource
+            Write-Host "    Creating $($Object.ObjectType) following the ARM Parameter File..."
+            $Result = New-AzResourceGroupDeployment -ResourceGroupName $AzResourceGroup -TemplateFile $Object.TemplatePath -TemplateParameterFile $Object.ParameterPath -Mode Incremental -ErrorAction SilentlyContinue -ErrorVariable objerror
+    
+            ## Sets a sleep of 10 seconds after object creation to allow Azure to update creation status, and mark as "running"
+            Start-Sleep -Seconds 10
+    
+            ## Verifies that no error occured when creating the Azure resource
+            If($objerror)
+            {
+                ## Creating the object following the ARM template failed.
+                Write-Host "      Error: Object failed to create.`n      $objerror" -ForegroundColor Red
+                Return
+            }
+            else 
+            {
+                ## Creating the object was successful
+                Write-Host "      $($Object.ObjectType) was successfully created."
+            }
+    
+            ## Copy GitHub Functions to newly created Azure Function
+            If($Object.ObjectType -eq "Function")
+            {
+                ## Gets the Azure Function Name from the Parameter JSON file contents.
+                $AzFunctionName = $jsonFunction.parameters.functionName.value
+    
+                ## Verifies the presence of the "CompiledFunctions.zip" file.
+                Write-Host "    Confirming Compiled Azure Functions is present"
+                IF(Test-Path $ArchiveFunctionZip)
+                {
+                    ## The "CompiledFunctions.zip" was found in the working directory
+                    Write-Host "      File Path Found: $ArchiveFunctionZip"
+
+                    ## Uploads the Windows Package Manager functions to the Azure Function.
+                    Write-Host "    Copying function files to the Azure Function."
+                    $Result = Publish-AzWebApp -ArchivePath $ArchiveFunctionZip -ResourceGroupName $AzResourceGroup -Name $AzFunctionName -Force
+                }
+                else 
+                {
+                    ## The "CompiledFunctions.zip" was not found. Unable to uploaded to the Azure Function.
+                    Write-Host "      File Path not found: $ArchiveFunctionZip" -ForegroundColor Red
+                }
+            }
+        }
+    }
+    End
+    {}
+}
+
+## Removes unsupported characters from the Resource Group Name
+$AzResourceGroup = $("$AzResourceGroup$Index").Replace("-","")
```

Comment:
Log to the user that something was changed so they are aware.

Reply:
Do any other fields need to be checked?

Reply:
Added an output to the PowerShell window that notifies the executing user that the hyphens were removed from the name.